azurerm/resource_arm_healthcare_service.go

```diff
@@ -0,0 +1,391 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/healthcareapis/mgmt/2018-08-20-preview/healthcareapis"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmHealthcareService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmHealthcareServiceCreateUpdate,
+		Read:   resourceArmHealthcareServiceRead,
+		Update: resourceArmHealthcareServiceCreateUpdate,
+		Delete: resourceArmHealthcareServiceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"kind": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "fhir",
+			},
+
+			"cosmosdb_throughput": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1000,
+			},
+
+			"access_policy_object_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.UUID,
+				},
+			},
+
+			"authentication_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 3,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"authority": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"audience": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"smart_proxy_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"cors_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 5,
```

Comment:
The code treats this as a singular property?

Reply:
done. Chaged to 1